### PR TITLE
Fix run automation

### PR DESCRIPTION
**Pull request by frenzymadness**
Issue described in #442

### CODE REVIEW

**Comment by frenzymadness** [comment]
This is now expected to fail because app.py is recognized as a python file and executed as is even APP_MODULE is defined to app:application and should be run via gunicorn.

**Comment by frenzymadness** [comment]
[test]

**Comment by frenzymadness** [comment]
The following failure is expected because the automation is currently broken. The automation in the run script tries to run the application as a Python script app.py even the `APP_MODULE` is set to `app:application` - but the file contains only definitions so the container finishes its run immediately. Next commit should fix it and hopefully don't break anything else.

```
14:46:26 [1m[INFO] Running test test_application ... [0m
14:46:26 
14:46:26 
14:46:26 [1m[INFO] Waiting for application container to start  ...[0m
14:46:26 
14:46:26 ---> Running application from Python script (app.py) ...
14:46:27 
14:46:27 [1m[INFO] Testing the image SCL enable[0m
14:46:27 
14:46:27 ERROR[exec /bin/bash -c python --version] Expected 'Python 3.8.', got 'Error response from daemon: No such container: f35f42967023445005909b656868f1054c12ffec53e5a314438cefde586d184f'
```

**Comment by frenzymadness** [comment]
[test][test-openshift]

**Comment by frenzymadness** [comment]
It seems that the last two commits fix the problem and the test added in the first two commits. So, ready for a review. Cc @pkubatrh @torsava

**Comment by phracek** [comment]
[test-openshift]

**Comment by phracek** [comment]
OpenShift 3.X does not start properly. But It should not block this PR.
LGTM from my point of view.

File: src/s2i/bin/run
Comment (pkubatrh) [inline src/s2i/bin/run:59]: We already know the variables are empty, so no need to check them again

```suggestion
  app_script_check=
  APP_SCRIPT="app.sh"
  app_file_check=
  APP_FILE="app.py"
```

Comment (frenzymadness) [reply src/s2i/bin/run]: Thank you!

**Comment by frenzymadness** [comment]
[test][test-openshift]

File: src/s2i/bin/run
Comment (pkubatrh) [reply src/s2i/bin/run]: Sources will need to be regenerated as well to get the changes to versioned files

**Comment by frenzymadness** [comment]
Rebased, regenerated, [test][test-openshift]

**Comment by frenzymadness** [comment]
[test][test-openshift]

**Comment by frenzymadness** [comment]
The tests have uncovered a real problem so I've tried to reimplement the logic in a simpler and more readable form.

[test][test-openshift]

**Comment by frenzymadness** [comment]
I have to fix this.

**Comment by frenzymadness** [comment]
[test][test-openshift]

**Comment by frenzymadness** [comment]
[test-openshift]

**Comment by frenzymadness** [comment]
Rebased [test][test-openshift][test-openshift-4]

**Comment by frenzymadness** [comment]
@phracek Tests fail on OS 4, do you know why?

This is now ready for review. I must admit that I've almost reimplemented the original functionality about the defaults but it's more readable IMO. The most important thing is that it solves the original issue and adds a test for it as well.

**Comment by phracek** [comment]
I will try it locally. Give me a time.
But the reason failure will be visible as soon as this pull request is merged https://github.com/sclorg/container-common-scripts/pull/196

**Comment by phracek** [comment]
Locally in OpenStack instance OpenShift tests are passing.

Let's try one more time. It looks like some network issues.

[test-openshift-4]

**Comment by phracek** [comment]
Locally in OpenStack instance OpenShift tests are passing.

Let's try one more time. It looks like some network issues.

[test-openshift-4]

**Review by phracek — APPROVED** [review]
LGTM

**Comment by frenzymadness** [comment]
Thank you!

**Pull request merged**
